tests/components/matrix/test_send_message.py

```diff
@@ -6,36 +6,118 @@
+                    ATTR_MESSAGE: "Test <b>html</b> message",
+                    ATTR_DATA: {ATTR_FORMAT: FORMAT_HTML},
+                    ATTR_TARGET: list(TEST_JOINABLE_ROOMS.keys()),
+                },
+                "expected_result": {
+                    "target_rooms": list(TEST_JOINABLE_ROOMS.keys()),
+                    "message_type": "m.room.message",
+                    "content": {
+                        "msgtype": "m.text",
+                        "body": "Test <b>html</b> message",
+                        "format": "org.matrix.custom.html",
+                        "formatted_body": "Test <b>html</b> message",
+                    },
+                },
+            },
+        ),
+        (
+            "Bot (notice) message",
+            {
+                "data": {
+                    ATTR_MESSAGE: "Test bot (notice) message",
+                    ATTR_DATA: {ATTR_FORMAT: FORMAT_NOTICE},
+                    ATTR_TARGET: list(TEST_JOINABLE_ROOMS.keys()),
+                },
+                "expected_result": {
+                    "target_rooms": list(TEST_JOINABLE_ROOMS.keys()),
+                    "message_type": "m.room.message",
+                    "content": {
+                        "msgtype": "m.notice",
+                        "body": "Test bot (notice) message",
+                    },
+                },
+            },
+        ),
+    ],
+)
 async def test_send_message(
-    hass: HomeAssistant, matrix_bot: MatrixBot, image_path, matrix_events, caplog
+    hass: HomeAssistant,
+    matrix_bot: MatrixBot,
+    image_path,
+    matrix_events,
+    caplog,
+    test_name,
+    test_params,
 ):
     """Test the send_message service."""
 
     await hass.async_start()
     assert len(matrix_events) == 0
     await matrix_bot._login()
 
-    # Send a message without an attached image.
-    data = {ATTR_MESSAGE: "Test message", ATTR_TARGET: list(TEST_JOINABLE_ROOMS)}
+    print(test_params)
     await hass.services.async_call(
-        MATRIX_DOMAIN, SERVICE_SEND_MESSAGE, data, blocking=True
+        MATRIX_DOMAIN, SERVICE_SEND_MESSAGE, test_params["data"], blocking=True
+    )
+
+    matrix_bot._handle_multi_room_send.assert_called_once_with(
+        **test_params["expected_result"]
     )
 
     for room_alias_or_id in TEST_JOINABLE_ROOMS:
         assert f"Message delivered to room '{room_alias_or_id}'" in caplog.messages
 
-    # Send an HTML message without an attached image.
+
+async def test_send_image(
+    hass: HomeAssistant, matrix_bot: MatrixBot, image_path, matrix_events, caplog
+):
+    """Test send a message with an attached image"""
+
+    await hass.async_start()
+    assert len(matrix_events) == 0
+    await matrix_bot._login()
+
     data = {
-        ATTR_MESSAGE: "Test message",
+        ATTR_MESSAGE: "Test <b>html</b> message",
         ATTR_TARGET: list(TEST_JOINABLE_ROOMS),
-        ATTR_DATA: {ATTR_FORMAT: FORMAT_HTML},
+        ATTR_DATA: {ATTR_IMAGES: [image_path.name]},
```

Comment:
Couldn't this be added as another parameterized option for `test_send_message`?

Reply:
Unfortunately I haven't find a way to check it using `matrix_bot._handle_multi_room_send.assert_called_once_with(...)` because the message content (example: `tmpeno21kcu`) is a random-generated string:
```
{'body': 'tmpeno21kcu', 'info': {'size': 144, 'mimetype': None, 'w': 50, 'h': 50}, 'msgtype': 'm.image', 'url': 'mxc://example.com/randomgibberish'}
```